cohorts/survival.py

```diff
@@ -145,41 +122,185 @@ def plot_kmf(df,
+             print_as_title=False):
+    """
+    Plot survival curves by splitting the dataset into two groups based on
+    condition_col. Report results for a log-rank test (if two groups are plotted)
+    or CoxPH survival analysis (if >2 groups) for association with survival.
+
+    Regarding definition of groups:
+        If condition_col is numeric, values are split into 2 groups.
+             - if threshold is defined, the groups are split on being > or < condition_col
+             - if threshold == 'median', the threshold is set to the median of condition_col
+        If condition_col is categorical or string, results are plotted for each unique value in the dataset.
+        If condition_col is None, results are plotted for all observations
+
+    Currently, if `strata_col` is given, the results are repeated among each stratum of the df.
+    A truly "stratified" analysis is not yet supported by may be soon.
+
+    Parameters
+    ----------
+        df: dataframe
+        condition_col: string, column which contains the condition to split on
+        survival_col: string, column which contains the survival time
+        censor_col: string,
+        strata_col: optional string, denoting column containing data to
+                    stratify by (default: None)
+        threshold: int or string, if int, condition_col is thresholded,
+                                  if 'median', condition_col thresholded
+                                  at its median
+                                  if 'median per-strata', & if stratified analysis
+                                  then condition_col thresholded by strata
+        title: Title for the plot, default None
+        ax: an existing matplotlib ax, optional, default None
+             note: not currently supported when `strata_col` is not None
+        with_condition_color: str, hex code color for the with-condition curve
+        no_condition_color: str, hex code color for the no-condition curve
+        with_condition_label: str, optional, label for TRUE condition case
+        no_condition_label: str, optional, label for FALSE condition case
+        color_map: dict, optional, mapping of hex-values to condition text
+          in the form of {value_name: color_hex_code}.
+          defaults to `sb.color_palette` using `default_color_palette` name,
+          or *_condition_color options in case of boolean operators.
+        label_map: dict, optional, mapping of labels to condition text.
+          defaults to "condition_name = condition_value", or *_condition_label
+          options in case of boolean operators.
+        color_palette: str, optional, name of sb.color_palette to use
+          if color_map not provided.
+        print_as_title: bool, optional, whether or not to print text
+          within the plot's title vs. stdout, default False
+    """
+    
+    # set reasonable default threshold value depending on type of condition_col
+    if threshold is None:
```

Comment:
Doesn't this duplicate logic in `plot_survival`? Should we move the `median` default out of there?